src/mlpack/methods/ann/layer/layer_types.hpp

```diff
@@ -282,6 +283,7 @@ using LayerTypes = boost::variant<
     LSTM<arma::mat, arma::mat>*,
     MaxPooling<arma::mat, arma::mat>*,
     MeanPooling<arma::mat, arma::mat>*,
+    LpPooling<arma::mat, arma::mat>*,
```

Comment:
Can you add the layer to `MoreTypes` (same file) instead and see if that works, we might run into some limitations here.

Reply:
I think it worked.

Reply:
Okay, great, once https://github.com/mlpack/mlpack/pull/2777 is ready this is going to be a lot easier.